File: Weapon.rb
require_relative 'CombatElement'

module Irrgarten

  # Esta clase representa las armas que utiliza el jugador en los ataques
  # durante los combates.
  #
  # @author Juan Caballero Santoyo


  class Weapon < CombatElement

   

    # Ejecuta un ataque utilizando el arma.
    #
    # Este método evalúa si el arma dispone de usos disponibles. Si es así, se decrementa
    # el contador de usos en 1 y se retorna la potencia asignada al arma. Si no quedan usos,
    # el ataque retorna 0, indicando que el arma no puede ser utilizada.
    #
    # @return [Integer] El valor del ataque, que corresponde a la potencia del arma o 0 si no hay usos.

    def attack
      self.produce_effect
    end

    # Retorna una representación en cadena del arma.
    #
    # El formato de la representación es "W[<potencia>, <usos>]", donde <potencia> y <usos>
    # son los atributos actuales del arma.
    #
    # @return [String] La representación textual del arma.

    def to_s
      return "W" + super
    end

    public_class_method :new
  end

end

File: java/src/irrgarten/Weapon.java

package irrgarten;

/**
 * Esta clase representa las armas que utiliza el jugador en los ataques
 * durante los combates.
 * 
 * @author Juan Caballero Santoyo
 */
public class Weapon extends CombatElement {
    
    
    /**
    * Constructor de la clase Weapon.
    * @param power El poder del arma.
    * @param uses El número de usos disponibles del arma.
    */
    public Weapon(float power, int uses){
        super(power,uses);
    }
    
    /**
     * Método para realizar una ataque con el arma.
     * @return La intensidad del ataque del arma.
    */
    public float attack(){
        return this.produceEffect();
    }
    
    /**
    * Método que devuelve una representación en forma de cadena de caracteres
    * del estado interno del objeto.
    * @return Representación en forma de cadena de caracteres del estado interno del objeto.
    */
    @Override
    public String toString(){
        return "W"+super.toString();
    }
    
}
